lib/msf/core/post/linux/system.rb

```diff
@@ -96,6 +96,33 @@ def get_sysinfo
     return system_data
   end
 
+  #
+  # return true if possible chroot environment detect, if not true then false
+  #
+  def is_chroot?
+    chroot_env = false
+
+    if not directory?("/proc/")
+      chroot_env = true
+    else
+      dir("/proc/").each do |pid|
+        pid = Integer(pid) rescue nil
+        if pid
+          filename = ::File.readlink("/proc/#{pid}/exe") rescue nil
```

Comment:
@hc0d3r You can use File.exists? to determine if the file exists. https://github.com/rapid7/metasploit-framework/blob/ea32c313d3596497be184161b784544100e0a3ec/lib/rex/post/meterpreter/extensions/stdapi/fs/file.rb#L187